script.js

```diff
@@ -0,0 +1,356 @@
+/* global SpatialNavigator */
+
+window.SmartScrollbox = (function(exports) {
+  'use strict';
+
+  var proto = Object.create(HTMLElement.prototype);
+
+  proto.createdCallback = function ssb_createdCallback() {
+
+    this._template = template.content.cloneNode(true);
+
+    this.listElem = this._template.getElementById('scrollbox-list');
+    this.createShadowRoot().appendChild(this._template);
+
+    // left and right margin of the scroll list
+    this.margin = this.hasAttribute('margin')?
```

Comment:
The hasAttribute('margin') is useless. We may use the following:
this.margin = this.getAttribute('margin') || 0;